src/sentry/spans/consumers/process/processor.py

```diff
@@ -0,0 +1,176 @@
+    snuba_span["start_timestamp_ms"] = int(start_timestamp.timestamp() * 1e3)
+    end_timestamp = datetime.utcfromtimestamp(relay_span["timestamp"])
+    snuba_span["duration_ms"] = max(
+        int((end_timestamp - start_timestamp).total_seconds() * 1e3),
+        0,
+    )
+
+    sentry_tags: MutableMapping[str, Any] = {}
+
+    if span_data:
+        for relay_tag, snuba_tag in TAG_MAPPING.items():
+            tag_value = span_data.get(relay_tag)
+            if tag_value is None:
+                if snuba_tag == "group":
+                    metrics.incr("spans.missing_group")
+            else:
+                sentry_tags[snuba_tag] = tag_value
+
+    if "op" not in sentry_tags and (op := relay_span.get("op", "")) is not None:
+        sentry_tags["op"] = op
+
+    if "status" not in sentry_tags and (status := relay_span.get("status", "")) is not None:
+        sentry_tags["status"] = status
+
+    if "status_code" in sentry_tags:
+        sentry_tags["status_code"] = int(sentry_tags["status_code"])
+
+    snuba_span["sentry_tags"] = sentry_tags
+
+    grouping_config = load_span_grouping_config()
+
+    if snuba_span["is_segment"]:
+        group = grouping_config.strategy.get_transaction_span_group(
+            {"transaction": span_data.get("transaction", "")},
+        )
+    else:
+        # Build a span with only necessary values filled.
+        span = Span(
+            op=snuba_span.get("op", ""),
+            description=snuba_span.get("description", ""),
+            fingerprint=None,
+            trace_id="",
+            parent_span_id="",
+            span_id="",
+            start_timestamp=0,
+            timestamp=0,
+            tags=None,
+            data=None,
+            same_process_as_parent=True,
+        )
+        group = grouping_config.strategy.get_span_group(span)
+
+    snuba_span["group_raw"] = group or "0"
+    snuba_span["span_grouping_config"] = {"id": grouping_config.id}
+
+    return snuba_span
+
+
+def _format_event_id(payload: Mapping[str, Any]) -> str:
+    event_id = payload.get("event_id")
+    if event_id:
+        return uuid.UUID(event_id).hex
+    return ""
+
+
+def _build_snuba_payload(snuba_span: Mapping[str, Any]) -> KafkaPayload:
+    snuba_payload = json.dumps(snuba_span).encode("utf-8")
+    return KafkaPayload(key=None, value=snuba_payload, headers=[])
+
+
+def _store_to_spanstore(snuba_span: Mapping[str, Any]) -> None:
+    spanstore.backend.set(snuba_span)
+
+
+def _decode_relay_span(message: Message[KafkaPayload]) -> Mapping[str, Any]:
+    payload = msgpack.unpackb(message.payload.value)
+    relay_span = payload["span"]
+    relay_span["project_id"] = payload["project_id"]
+    relay_span["event_id"] = _format_event_id(payload)
+    return relay_span
+
+
+class SpansProcessor:
+    def __init__(
+        self,
+        bigtable_project_id: str,
+        bigtable_instance_id: str,
+        bigtable_table_id: str,
```

Comment:
We should probably not provide these parameters here. Instead we can rely on the `SENTRY_SPANSTORE` setting and the corresponding `SENTRY_SPANSTORE_OPTIONS` to populate spanstore specific configurations. That way the solution works for self hosted as well.